### PR TITLE
LVM and general update

### PR DESCRIPTION
**Pull request by Jeroen0494**
Hi,

* Add profile for LVM and update-secureboot-policy
* Various fixes all over

cni-calico has 2 new variables, date and time. These could potentially be used for other profiles as well. Should I move them to tunables?

### CODE REVIEW

File: apparmor.d/profiles-a-f/blkdeactivate [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# apparmor.d - Full set of apparmor profiles
+# Copyright (C) 2022 Jeroen Rijken
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = /{usr/,}sbin/blkdeactivate
```
Comment (roddhjav) [inline apparmor.d/profiles-a-f/blkdeactivate:9]:
```suggestion
@{exec_path} = /{usr/,}{s,}bin/blkdeactivate
```

File: apparmor.d/profiles-a-f/blkdeactivate [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+profile blkdeactivate @{exec_path} flags=(complain) {
+  include <abstractions/base>
+  include <abstractions/nameservice-strict>
+
+  @{exec_path} rm,
+  /{usr/,}sbin/dmsetup rPUx,
```
Comment (roddhjav) [inline apparmor.d/profiles-a-f/blkdeactivate:15]:
```suggestion
  /{usr/,}{s,}bin/dmsetup rPUx,
```

File: apparmor.d/profiles-a-f/blkdeactivate [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  /{usr/,}bin/grep rix,
+  /{usr/,}bin/lsblk rPx,
+  /{usr/,}sbin/lvm rPx,
```
Comment (roddhjav) [inline apparmor.d/profiles-a-f/blkdeactivate:18]:
```suggestion
  /{usr/,}{s,}bin/lvm rPx,
```

File: apparmor.d/profiles-a-f/dmeventd [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# apparmor.d - Full set of apparmor profiles
+# Copyright (C) 2022 Jeroen Rijken
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = /{usr/,}sbin/dmeventd
```
Comment (roddhjav) [inline apparmor.d/profiles-a-f/dmeventd:9]:
```suggestion
@{exec_path} = /{usr/,}{s,}bin/dmeventd
```

File: apparmor.d/profiles-g-l/lvm [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# apparmor.d - Full set of apparmor profiles
+# Copyright (C) 2022 Jeroen Rijken
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = /{usr/,}sbin/lvm
```
Comment (roddhjav) [inline apparmor.d/profiles-g-l/lvm:9]:
```suggestion
@{exec_path} = /{usr/,}{s,}bin/lvm
```

File: apparmor.d/profiles-g-l/lvmconfig [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# apparmor.d - Full set of apparmor profiles
+# Copyright (C) 2022 Jeroen Rijken
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = /{usr/,}sbin/lvmconfig
```
Comment (roddhjav) [inline apparmor.d/profiles-g-l/lvmconfig:9]:
```suggestion
@{exec_path} = /{usr/,}{s,}bin/lvmconfig
```

File: apparmor.d/profiles-g-l/lvmdump [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# apparmor.d - Full set of apparmor profiles
+# Copyright (C) 2022 Jeroen Rijken
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = /{usr/,}sbin/lvmdump
```
Comment (roddhjav) [inline apparmor.d/profiles-g-l/lvmdump:9]:
```suggestion
@{exec_path} = /{usr/,}{s,}bin/lvmdump
```

File: apparmor.d/profiles-g-l/lvmpolld [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# apparmor.d - Full set of apparmor profiles
+# Copyright (C) 2022 Jeroen Rijken
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = /{usr/,}sbin/lvmpolld
```
Comment (roddhjav) [inline apparmor.d/profiles-g-l/lvmpolld:9]:
```suggestion
@{exec_path} = /{usr/,}{s,}bin/lvmpolld
```

**Comment by roddhjav** [comment]
Thank for this PR.

> cni-calico has 2 new variables, date and time. These could potentially be used for other profiles as well. Should I move them to tunables?

That can be usefull elsewhere, so yes, put them in `apparmor.d/tunables/extend`

**Comment by Jeroen0494** [comment]
Suggestions applied.

File: apparmor.d/profiles-a-f/dkms [hunk 9]
```diff
@@ -23,47 +23,47 @@ profile dkms @{exec_path} flags=(attach_disconnected) {
...
-  /{usr/,}bin/wc         rix,
-  /{usr/,}bin/rmdir      rix,
+  /{usr/,}bin/echo       rix,
   /{usr/,}bin/find       rix,
-  /{usr/,}bin/{,e}grep   rix,
-  /{usr/,}bin/{,g,m}awk  rix,
-  /{usr/,}bin/cp         rix,
-  /{usr/,}bin/date       rix,
+  /{usr/,}bin/getconf    rix,
+  /{usr/,}bin/grep       rix,
```
Comment (nobodysu) [inline apparmor.d/profiles-a-f/dkms:34]: Lines 34 and 55 are mutually exclusive.

Comment (Jeroen0494) [reply apparmor.d/profiles-a-f/dkms]: Thanks, fixed.

**Comment by Jeroen0494** [comment]
Rebased on master.

File: apparmor.d/profiles-g-l/lvm [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+profile lvm @{exec_path} flags=(complain) {
+  include <abstractions/base>
+  include <abstractions/consoles>
+  include <abstractions/dbus-strict>
+  include <abstractions/disks-write>
+
+  capability sys_admin,
+  capability sys_nice,
+  deny capability net_admin,
```
[the remark this answers is not in the document]
Comment (roddhjav) [reply apparmor.d/profiles-g-l/lvm:18]: It seems it is required for multipath support. Btw, `net_admin` does not give network access, it allow to change network settings.

[the remark this answers is not in the document]
Comment (Jeroen0494) [reply apparmor.d/profiles-g-l/lvm]: Alright, I've included it.

**Comment by roddhjav** [comment]
Thanks, Merged!

**Pull request merged**